cmd/dagger/main.go

```diff
@@ -345,6 +346,34 @@ func toUpperBold(s string) string {
 	return termenv.String(upperCase).Bold().String()
 }
 
+// separateAndModifyFlags separates optional flags from required flags
+func separateAndModifyFlags(originalFlags *pflag.FlagSet) *pflag.FlagSet {
+	mergedFlags := pflag.NewFlagSet("merged", pflag.ContinueOnError)
+	mergedFlags.SortFlags = false
+
+	optionalFlags := pflag.NewFlagSet("optional", pflag.ContinueOnError)
+
+	// separate optional flags from required flags
+	originalFlags.VisitAll(func(flag *pflag.Flag) {
+		clone := *flag
+
+		// Append [required] and show required flags first
+		if flag.Annotations["cobra_annotation_bash_completion_one_required_flag"] != nil {
```

Comment:
Better to use the const:
```suggestion
		if flag.Annotations[cobra.BashCompOneRequiredFlag] != nil {
```